graphql_client_cli/src/error.rs

```diff
@@ -28,25 +26,16 @@ impl Error {
 }
 
 // This is the impl that shows up when the error bubbles up to `main()`.
-impl Debug for Error {
+impl std::fmt::Debug for Error {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        if let Some(msg) = &self.message {
-            f.write_str(msg)?;
-            f.write_str("\n")?;
-        }
-
-        if self.source.is_some() && self.message.is_some() {
-            f.write_str("Cause: ")?;
+        match (&self.message, &self.source) {
+            (Some(msg), Some(source)) => {
+                write!(f, "{msg}\nCause: {source}\nLocation: {}", self.location)
+            }
+            (Some(msg), None) => write!(f, "{msg}\nLocation: {}", self.location),
+            (None, Some(source)) => write!(f, "{source}\nLocation: {}", self.location),
+            (None, None) => write!(f, "\nLocation: {}", self.location),
```

Comment:
@tomhoule i've copied the original implementation here, but i suspect the leading newline here is not intentional?

Reply:
Probably not. Ideally we'd have tests for the display impl.

Reply:
I can add those before this lands